### PR TITLE
Bugfixes for gated and validate commands

### PR DESCRIPTION
**Pull request by ralsina**
Fix LP: [#1629991](https://bugs.launchpad.net/snapcraft/+bug/1629991) with:

* Better handling of snaps with no validations in the gated command
* Minor internal fixes in the validate command

### CODE REVIEW

File: integration_tests/test_gated.py [hunk 1]
```diff
@@ -42,3 +42,9 @@ def test_gated_unknown_snap_failure(self):
         self.login()
         self.assertEqual(1, self.gated(
             'unknown', expected_error="Snap 'unknown' was not found"))
+
+    def test_gated_no_validations(self):
+        self.addCleanup(self.logout)
+        self.login()
+        self.assertEqual(0, self.gated(
+            'basic', expected_error="No validations for snap 'basic'"))
```
Comment (cprov) [inline integration_tests/test_gated.py]: Roberto,  "expected_error" sounds misleading. This text is not an error, right ?

Comment (cprov) [inline integration_tests/test_gated.py]: Also, the PR should mention LP: #1629991

Comment (ralsina) [inline integration_tests/test_gated.py]: Indeed that's not an error, I'll rename to expected_output.

**Comment by kyrofa** [comment]
@ralsina it's handy to have the PR reference the bug, thank you, but we really need it in a commit as well (refer to the [contribution guide](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md) for more information).

**Comment by kyrofa** [comment]
Also, as the PR title by default turns into the commit title, it'd be better to have the buglink in the PR description instead of the title.

**Comment by ralsina** [comment]
@kyrofa will fix right away, thanks!

File: snapcraft/_store.py [hunk 2]
```diff
@@ -385,27 +385,25 @@ def gated(snap_name):
...
+    if validations:
+        table_data = []
+        for v in validations:
+            table_data.append([v['approved-snap-name'],
+                               v['approved-snap-revision']])
+        tabulated = tabulate(table_data, headers=['Name', 'Approved'],
+                             tablefmt="plain")
+        print(tabulated)
+    else:
+        print('No validations for snap {!r}'.format(snap_name))
```
Comment (ralsina) [inline snapcraft/_store.py]: ack, changing.

File: integration_tests/test_gated.py [hunk 3]
```diff
@@ -35,10 +35,16 @@ def test_gated_success(self):
 
     def test_gated_no_login_failure(self):
         self.assertEqual(1, self.gated(
-            'ubuntu-core', expected_error='Have you run "snapcraft login'))
+            'ubuntu-core', expected_output='Have you run "snapcraft login'))
```
Comment (sergiusens) [inline integration_tests/test_gated.py]: Is there a " mismatch in the actual code?

File: snapcraft/_store.py (hunk 2 above)
Comment (sergiusens) [inline snapcraft/_store.py]: did this change?

File: integration_tests/test_gated.py (hunk 3 above)
Comment (ralsina) [inline integration_tests/test_gated.py]: No, that's not a mismatch, scared me for a bit :-)

The error message has double quotes, the expected output has single quotes, and is not the whole message.

File: snapcraft/_store.py (hunk 2 above)
Comment (ralsina) [inline snapcraft/_store.py]: Yes, changed it.

**Comment by elopio** [comment]
integration_tests/test_gated.py:50:80: E501 line too long (82 > 79 characters)

**Comment by ralsina** [comment]
Oops, sorry @elopio forgot to check for lint, fixed now.

**Comment by ralsina** [comment]
Hold the merge for a second, coveralls was right and there is a missing test for using the --key-name parameter (well, it was also missing before, but hey, I added it)

**Comment by sergiusens** [comment]
oh, you have conflicts!

**Comment by ralsina** [comment]
Not anymore!

**Pull request merged**